sycl/doc/extensions/AsyncGroupCopy/async_group_copy.asciidoc

```diff
@@ -0,0 +1,157 @@
+used by permission by Khronos.
+
+This extension is written against the SYCL 2020 revision 4 specification.  All
+references below to the "core SYCL specification" or to section numbers in the
+SYCL specification refer to that revision.
+
+## Contributors
+
+* Gordon Brown
+* Tadej Ciglaric
+* Jack Kirk
+* Finlay Marno
+
+## Introduction
+
+This document proposes `async_group_copy`, which will generalize
+`async_work_group_copy` to work with sub-groups as well as work-groups.
+A `wait_for` function will also be also be added to block until the memory
+copy is complete and to ensure memory consistency for the group, this will use
+the same groups as `async_group_copy`.
+
+## Feature test macro
+
+This extension provides a feature-test macro as described in the core SYCL
+specification section 6.3.3 "Feature test macros". Therefore, an implementation
+supporting this extension must predefine the macro
+SYCL_EXT_ONEAPI_ASYNC_GROUP_COPY to one of the values defined in the table
+below. Applications can test for the existence of this macro to determine if the
+implementation supports this feature, or applications can test the macro’s value
+to determine which of the extension’s APIs the implementation supports.
+
+[%header,cols="1,5"]
+|===
+|Value |Description
+|1     |Initial extension implementation.
+|===
+
+
+## New `async_group_copy` function.
+`async_group_copy` will work very similarly to `async_work_group_copy`, but
+with the ability to utilize individual sub-groups. The group will be provided
+as a template parameter, in the same fashion as `group_barrier`.
+`async_group_copy` is intended as a potential optimization, and may add more
+overhead than a simple synchronous implementation in some cases.
+
+`async_group_copy` will asynchronously copy a given number of values of type
+`dataT` from global memory to local memory or from local memory to global
+memory. If an asynchorous copy is not supported then `async_group_copy` will
+fall back on a synchronous copy. `async_group_copy` is a free function with the
+following properties:
+
+* The work-items will not necessarily be synchronized, so must have a
+  consistent view of the memory before the invocation.
+* The order of operations is not specified.
+* All work-items of the group are required to call the function with the same
+  arguments, otherwise the behaviour is undefined.
+* All work-items of the group are required to call the function in convergent
+  control flow, otherwise the behaviour is undefined.
+* The type of `dataT` should be a scalar or vector type.
+* The data written by `async_group_copy` should not be read until `wait_for`
+  has been called on the returned `device_event`.
+
+The structs `dest_stride` and `src_stride` have been introduced to reduce errors
+when specifying the stride.
+
+Note that the destination and source arguments have been swapped to be more in
+line with other SYCL copy functions.
```

Comment:
```suggestion
NOTE: The order of the destination and source arguments differs from the SYCL 1.2.1 `async_work_group_copy` function to be more in line with other copy functions in SYCL 2020.
```

I don't think it's clear what has been swapped here.  I suggest making it a non-normative note and making the comparison explicit.